pkg/manifest/mantaray.go

```diff
@@ -124,6 +143,53 @@ func (m *mantarayManifest) Store(ctx context.Context, mode storage.ModePut) (swa
 	return address, nil
 }
 
+func (m *mantarayManifest) IterateAddresses(ctx context.Context, fn swarm.AddressIterFunc) error {
+	reference := swarm.NewAddress(m.trie.Reference())
+
+	if swarm.ZeroAddress.Equal(reference) {
+		return ErrMissingReference
+	}
+
+	walker := func(path []byte, node *mantaray.Node, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if node != nil {
+			var stop bool
+
+			if node.Reference() != nil {
+				ref := swarm.NewAddress(node.Reference())
+
+				stop = fn(ref)
+				if stop {
+					return errStopIterator
+				}
+			}
+
+			if node.IsValueType() && node.Entry() != nil {
+				entry := swarm.NewAddress(node.Entry())
+				stop = fn(entry)
+				if stop {
+					return errStopIterator
```

Comment:
just  return nil here and then just

     return  m.trie.WalkNode([]byte{}, m.loader, walker)

no?

Reply:
The reason for this flow was because this code is using some other library for iteration (i.e. `manifest` project).
Implementation(s) in the `manifest` project for "walker" function were modeled according to Go `filepaht.WalkFunc`, which does not have some easy way to "stop" iteration (as far as I remember). Only way is to return error from (this) implemented "walker" function, which will be propagated to code here. Because of this, internal error is retuned to stop iteration, which is ignored later on.

This is in contrast to the `swarm.AddressIterFunc`, which is more modeled in the line with some DB iteration functions.